clusterloader2/pkg/measurement/common/profile.go

```diff
@@ -218,6 +221,17 @@ func (p *profileMeasurement) shouldExposeApiServerDebugEndpoint() bool {
 	return p.config.componentName == "kube-apiserver"
 }
 
+func shouldGetAPIServerByK8sClient(componentName string) bool {
+	// In some cases the kube-apiserver cannot be reached by curl.
+	// We add a config here as a walkaround.
+	getAPIServerByK8sClient, err := strconv.ParseBool(os.Getenv("GET_APISERVER_PPROF_BY_K8S_CLIENT"))
+	if err != nil {
+		klog.Warning("GET_APISERVER_PPROF_BY_K8S_CLIENT not set, using curl by default")
```

Comment:
If errror is returned it doesn't imply that the GET_APISERVER_PPROF_BY_K8S_CLIENT wasn't set